api/context/src/main/java/io/opentelemetry/context/propagation/MultiTextMapPropagator.java

```diff
@@ -45,13 +45,22 @@
 
   @Override
   public <C> void inject(Context context, @Nullable C carrier, TextMapSetter<C> setter) {
+    if (context == null || setter == null) {
+      return;
+    }
     for (TextMapPropagator textPropagator : textPropagators) {
       textPropagator.inject(context, carrier, setter);
     }
   }
 
   @Override
   public <C> Context extract(Context context, @Nullable C carrier, TextMapGetter<C> getter) {
+    if (context == null) {
+      return Context.root();
+    }
+    if (getter == null) {
+      return context;
+    }
     for (TextMapPropagator textPropagator : textPropagators) {
       context = textPropagator.extract(context, carrier, getter);
     }
```

Comment:
As said [above](https://github.com/open-telemetry/opentelemetry-java/pull/2939/files#r583761460), I would not guard against null before this loop also here in extract, but only ensure that we don't return `null` even if `null` was passed in as `context` (and `textPropagators` was possibly empty).